src/BalanceTx.purs

```diff
@@ -109,22 +103,46 @@ import Undefined (undefined)
 -- These derivations may need tweaking when testing to make sure they are easy
 -- to read, especially with generic show vs newtype show derivations.
 data BalanceTxFailure
-  = UtxosAtFailure' UtxosAtFailure
+  = GetWalletAddressFailure' GetWalletAddressFailure
+  | GetWalletCollateralFailure' GetWalletCollateralFailure
+  | UtxosAtFailure' UtxosAtFailure
+  | UtxoIndexToUtxoFailure' UtxoIndexToUtxoFailure
   -- | BalanceTxMFailure' BalanceTxMFailure -- REMOVE OR CHANGE THIS
   | ReturnAdaChangeFailure' ReturnAdaChangeFailure
   | AddTxCollateralsFailure' AddTxCollateralsFailure
   | ToEitherTransactionInputFailure' ToEitherTransactionInputFailure
   | BalanceTxInsFailure' BalanceTxInsFailure
   | BalanceNonAdaOutsFailure' BalanceNonAdaOutsFailure
-  -- | AddSignatoriesFailure' AddSignatoriesFailure -- REMOVE OR CHANGE THIS
+  | SignTxFailure' SignTxFailure
   | CalculateMinFeeFailure' CalculateMinFeeFailure
   | BuildTxRawFailure' BuildTxRawFailure
+  | ImpossibleFailure' Impossible
 
 derive instance genericBalanceTxFailure :: Generic BalanceTxFailure _
 
 instance showBalanceTxFailure :: Show BalanceTxFailure where
   show = genericShow
 
+newtype GetWalletAddressFailure = GetWalletAddressFailure GetWalletAddressFailureReason
+
+derive instance newtypeGetWalletAddressFailure :: Newtype GetWalletAddressFailure _
+derive newtype instance showGetWalletAddressFailure :: Show GetWalletAddressFailure
+
+data GetWalletAddressFailureReason = CouldNotGetNamiWalletAddress
+
+instance showGetWalletAddressFailureReason :: Show GetWalletAddressFailureReason where
+  show CouldNotGetNamiWalletAddress = "Couldn't get Nami wallet address."
```

Comment:
I think it's best to treat `Show` as something that prints PS expressions that can be evaluated and not arbitrarily descriptive `String`s. If you'd like to generate descriptions of errors, I would suggest something like `Affjax.printError`